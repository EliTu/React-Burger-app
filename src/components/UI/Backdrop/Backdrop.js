import React from 'react';
import styles from './Backdrop.module.css';
import PropTypes from 'prop-types';

const Backdrop = ({ show, removeBackdrop }) => {
	// CSS Modules styles:
	const { Backdrop } = styles;

	return show && <div className={Backdrop} onClick={removeBackdrop} />;
};

Backdrop.propTypes = {
	show: PropTypes.bool,
	removeBackdrop: PropTypes.func,
};

export default Backdrop;
