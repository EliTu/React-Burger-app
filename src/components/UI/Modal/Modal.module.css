@import '../../../variables.css';

.Modal {
    position: fixed;
    z-index: 500;
    margin: auto 0;
    color: var(--coloring-dark);
    width: 90%;
    border-radius: 8px;
    outline: 2px solid var(--coloring-opaque-dark);
    outline-offset: -13px;
    box-shadow: var(--shadow-dark);
    padding: 2rem;
    left: 5%;
    top: 20%;
    box-sizing: border-box;
    transition: all 0.4s ease-out;
    text-align: center;
    background-color: var(--primary-background-light);
}

/* large screen */
@media (min-width: 600px) {
    .Modal {
        width: 500px;
        left: calc(50% - 250px);
    }
}