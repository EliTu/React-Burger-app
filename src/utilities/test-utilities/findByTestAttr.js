// Used to find a specific DOM element that is specified with the 'data-test' attribute to better traverse while testing:
export const findByTestAttr = (component, attr) => {
	const wrapper = component.find(`[data-test='${attr}']`);
	return wrapper;
};
