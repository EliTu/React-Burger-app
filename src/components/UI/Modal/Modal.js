import React from 'react';
import styles from './Modal.module.css';
import Backdrop from '../Backdrop/Backdrop';
import PropTypes from 'prop-types';

const Modal = ({ children, show, closeModalHandler }) => {
	// CSS Modules styles:
	const { Modal } = styles;

	// Set conditional inline styles:
	const translateModal = show ? 'translateX(0)' : 'translateX(100vh)';
	const adjustOpacity = show ? '1' : '0';

	return (
		<>
			<Backdrop show={show} removeBackdrop={closeModalHandler} />
			<div
				className={Modal}
				data-test="wrapper-div"
				style={{
					transform: translateModal,
					opacity: adjustOpacity,
				}}
			>
				{children}
			</div>
		</>
	);
};

Modal.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node,
	]),
	show: PropTypes.bool,
	closeModalHandler: PropTypes.func,
};

export default React.memo(Modal);
