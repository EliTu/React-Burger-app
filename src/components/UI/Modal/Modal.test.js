import React from 'react';
import Backdrop from '../Backdrop/Backdrop';
import { findByTestAttr } from '../../../utilities/test-utilities/findByTestAttr';
import { shallow } from 'enzyme';
import Modal from './Modal';

describe('<Modal />', () => {
	// const component = setShallowAsComponent(<Modal />);
	let component;
	beforeEach(() => {
		component = shallow(<Modal />);
	});

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component).toBeTruthy();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should render the Backdrop component if mounted', () => {
		expect(component).toMatchSnapshot();
		expect(component.find(Backdrop)).toBeTruthy();
		expect(component.find(Backdrop)).toHaveLength(1);
		expect(component.find(Backdrop)).not.toHaveLength(2);
	});

	it('should render the wrapper <div> containing the children', () => {
		const wrapperDiv = findByTestAttr(component, 'wrapper-div');
		expect(component).toMatchSnapshot();
		expect(wrapperDiv).toBeTruthy();
		expect(wrapperDiv.length).toBe(1);
		expect(wrapperDiv.length).not.toBe(2);
	});

	it('should render the children props without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.children()).toBeTruthy();
		expect(component.children()).not.toHaveLength(0);
	});
});
