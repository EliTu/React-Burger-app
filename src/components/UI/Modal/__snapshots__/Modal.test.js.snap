// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render the Backdrop component if mounted 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="Modal"
    data-test="wrapper-div"
    style={
      Object {
        "opacity": "0",
        "transform": "translateX(100vh)",
      }
    }
  />
</Fragment>
`;

exports[`<Modal /> should render the children props without errors 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="Modal"
    data-test="wrapper-div"
    style={
      Object {
        "opacity": "0",
        "transform": "translateX(100vh)",
      }
    }
  />
</Fragment>
`;

exports[`<Modal /> should render the wrapper <div> containing the children 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="Modal"
    data-test="wrapper-div"
    style={
      Object {
        "opacity": "0",
        "transform": "translateX(100vh)",
      }
    }
  />
</Fragment>
`;

exports[`<Modal /> should render without errors 1`] = `
<Fragment>
  <Backdrop />
  <div
    className="Modal"
    data-test="wrapper-div"
    style={
      Object {
        "opacity": "0",
        "transform": "translateX(100vh)",
      }
    }
  />
</Fragment>
`;
